### ISSUE TITLE
Group node widgets reverting to default

### ISSUE THREAD

**Issue by antrobot1234** [comfyanonymous/ComfyUI#2771]
If you hide a widget of a group node, the value that is passed in will revert back to the value the group node had on creation.
![image](https://github.com/comfyanonymous/ComfyUI/assets/21179158/63f9f9e8-d137-4597-a814-a9bcd4585aca)
![image](https://github.com/comfyanonymous/ComfyUI/assets/21179158/adcdc0be-b131-48ad-9dc6-c0e98d9ee9c8)

**Comment by devingfx** [comment]
Yeah, the same happen if you convert a groupnode widget to an input, the value of the original node (inside the groupnode) widget is used (not the external linked one)
